tests/e2e/features/presentationViewer.feature

```diff
@@ -3,11 +3,19 @@ Feature: markdown presentation viewer
   I want to view markdown documents in presentation format
   So that I can easily present content in a structured manner
   
-  Background:
-    Given user "admin" has uploaded the markdown file "test-markdown.md" using API
-    And user "admin" has logged in
 
+  Scenario Outline: validate content of markdown file in presentation viewer
+    Given user "admin" has uploaded the markdown file "test-markdown.md" with content "<markdown-content>" using API
+    And user "admin" has logged in
```

Comment:
```suggestion
    Given user "admin" has logged in
    And user "admin" has uploaded the markdown file "test-markdown.md" with content "<markdown-content>" using API
```

Is not first step have to admin login before upload ?

Reply:
Since there is not going on sequence of actions. Someone unknown about this things they will be just in confusion.

Just because page have to refresh Is it good practice to swap the steps @anon-pradip @saw-jan

Reply:
`And user "admin" has logged in`
this is a step where admin logs in "to the web UI".
It is not the admin "logging in" to the API, or some other "underneath" login thing.

Maybe it is clearer for the words of this step to be something like:
`And user "admin" has logged in to the web UI`

Then it will not be confusing to see the upload via API happen before login.

Reply:
I think it's also possible to use passive tense in Given steps, where you don't have to specify the user if this doesn't matter for the scenario:
Given the markdown file "test-markdown.md" with content "<markdown-content>" has been upladed.

Reply:
> I think it's also possible to use passive tense in Given steps, where you don't have to specify the user if this doesn't matter for the scenario: Given the markdown file "test-markdown.md" with content "" has been uploaded.

but user "admin" is passed to other functions too.

Reply:
> `And user "admin" has logged in` this is a step where admin logs in "to the web UI". It is not the admin "logging in" to the API, or some other "underneath" login thing.
> 
> Maybe it is clearer for the words of this step to be something like: `And user "admin" has logged in to the web UI`
> 
> Then it will not be confusing to see the upload via API happen before login.

If we add "to the web UI" then I think we should add the same to other steps too; like:
`When the user previews the markdown file in the web UI`

Reply:
Here is a random example scenario from the oC10 test suite:
```
  Scenario: mark a file as favorite and list it in favorites page
    Given user "Alice" has uploaded file "filesForUpload/data.zip" to "/data.zip"
    And user "Alice" has uploaded file "filesForUpload/lorem.txt" to "/lorem.txt"
    And user "Alice" has uploaded file with content "file with comma" to "s,a,m,p,l,e.txt"
    And user "Alice" has logged in using the webUI
    And the user has browsed to the files page
    When the user marks file "data.zip" as favorite using the webUI
    And the user marks file "s,a,m,p,l,e.txt" as favorite using the webUI
    Then file "data.zip" should be marked as favorite on the webUI
    And file "s,a,m,p,l,e.txt" should be marked as favorite on the webUI
    When the user reloads the current page of the webUI
    Then file "data.zip" should be marked as favorite on the webUI
    And file "s,a,m,p,l,e.txt" should be marked as favorite on the webUI
    And file "data.zip" should be listed in the favorites page on the webUI
    And file "lorem.txt" should not be listed in the favorites page on the webUI
```

We keep mentioning "using the webUI", "on the webUI" in lots of steps. Partly that is because that set of test suites also has test suites/scenarios that explicitly test the API, without the webUI. And so we wanted to be really sure which interface was being used to do actions, and to check results.

But in the "web e2e" tests, we mention "using API" for steps where we want to to setup with the API, but don't bother to mention "using the UI" for the actual When steps that drive the UI, or for the Then steps that check things that are on the web UI.

So I am thinking do just have the login step mention the web UI. After that, I think it is pretty obvious that the following steps are things that happen on a UI.